### PR TITLE
ROX-9476: Change start behavior of the Sensor components

### PR DESCRIPTION
**Pull request by kovayur**
## Description

Change startup behavior of the sensor components.

Current behavior: log error, not panic
Expected behavior: panic in development build, log in release build (using #1444)

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

## Testing Performed
Unit tests

### CODE REVIEW

**Review by SimonBaeumer — APPROVED** [review]

File: sensor/kubernetes/localscanner/tls_issuer_test.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func TestLocalScannerTLSIssuerStartAlreadyStartedFailure(t *testing.T) {
 	secondStartErr := fixture.tlsIssuer.Start()
 
 	assert.NoError(t, startErr)
-	assert.NoError(t, secondStartErr)
+	assert.Error(t, secondStartErr)
```
Comment (SimonBaeumer) [inline sensor/kubernetes/localscanner/tls_issuer_test.go:148]:
```suggestion
	require.Error(t, secondStartErr)
```

File: sensor/kubernetes/localscanner/tls_issuer_test.go [hunk 2]
```diff
@@ -131,7 +131,7 @@ func TestLocalScannerTLSIssuerRefresherFailureStartFailure(t *testing.T) {
 
 	startErr := fixture.tlsIssuer.Start()
 
-	assert.NoError(t, startErr)
+	assert.Error(t, startErr)
```
Comment (SimonBaeumer) [inline sensor/kubernetes/localscanner/tls_issuer_test.go:134]: Typically more assertions are not necessary if an error occurred every following assertion fails.
```suggestion
	require.Error(t, startErr)
```

**Review by porridge — APPROVED** [review]

File: sensor/kubernetes/localscanner/tls_issuer_test.go (hunk 1 above)
Comment (kovayur) [reply sensor/kubernetes/localscanner/tls_issuer_test.go]: Fixed.

File: sensor/kubernetes/localscanner/tls_issuer_test.go (hunk 2 above)
Comment (kovayur) [reply sensor/kubernetes/localscanner/tls_issuer_test.go]: Thanks for advise, fixed.

**Review by juanrh — APPROVED** [review]

**Pull request merged**